modules/core/src/components/log-viewer/core-3d-viewer.js

```diff
@@ -263,10 +260,9 @@ export default class Core3DViewer extends PureComponent {
           // Check lookAheads first because it will contain the selected futures
           // while streams would contain the full futures array
           const stream = lookAheads[streamName] || streams[streamName];
-          const streamMetadata = getStreamMetadata(metadata, streamName);
           const coordinateProps = resolveCoordinateTransform(
             frame,
-            streamMetadata,
+            streamMetadata[streamName],
```

Comment:
It's a nitpick, but I would rename `streamMetadata` to `streamsMetadata` across the PR, just because it's the metadata map of all streams, not just one stream.  And so here `streamMetadata = streamsMetadata[streamName]`